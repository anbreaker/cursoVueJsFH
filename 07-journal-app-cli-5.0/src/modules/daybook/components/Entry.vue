<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: 10 } })"
  >
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">09</span>
      <span class="mx-1 fs-5">June</span>
      <span class="mx-2 fw-light">2022, Thursday</span>
    </div>

    <div class="entry-description">
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio ad itaque voluptatem,
      exercitationem amet nam hic, neque culpa ex sint aliquid! Dolorum dolore nisi quis
      eveniet nobis, voluptatum excepturi nostrum.
    </div>
  </div>
</template>

<script>
  export default {};
</script>

<style lang="scss" scoped>
  .entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2 ease-in;

    &:hover {
      background-color: lighten($color: black, $amount: 30);
    }

    .entry-description {
      font-size: 0.8rem;
    }
  }
</style>
