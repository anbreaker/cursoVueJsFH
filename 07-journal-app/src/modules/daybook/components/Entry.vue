<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: entry.id } })"
  >
    <!-- Title -->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 bold">{{ day }}</span>
      <span class="mx-1 fs-5">{{ month }}</span>
      <span class="mx-2 fw-light">{{ yearDay }}</span>
    </div>

    <div class="entry-description">
      {{ shortText }}
    </div>
  </div>
</template>

<script>
// prettier-ignore
const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
const days   = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'] // prettier-ignore

export default {
  props: {
    entry: {
      type: Object,
      required: true
    }
  },
  computed: {
    shortText() {
      return this.entry.text.length > 130
        ? this.entry.text.substring(0, 130) + "..."
        : this.entry.text;
    },

    // TODO refactor this functions
    day() {
      const date = new Date(this.entry.date);

      return date.getDate();
    },

    month() {
      const date = new Date(this.entry.date);

      return months[date.getMonth()];
    },

    yearDay() {
      const date = new Date(this.entry.date);

      return `${date.getFullYear()}, ${days[date.getDay()]}`;
    }
  }
};
</script>

<style lang="scss" scoped>
</style>