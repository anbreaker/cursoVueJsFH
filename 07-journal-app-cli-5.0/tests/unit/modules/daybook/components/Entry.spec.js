import { shallowMount } from '@vue/test-utils';

import Entry from '@/modules/daybook/components/Entry.vue';

import { journalState } from '../../../../../tests/mock-data/test-journal-state';

describe('testing Entry Component', () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(Entry, {
    props: {
      entry: journalState.entries[0],
    },

    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  test('should be match with the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('click on the button should redirect to "entry-container".', () => {
    const entryContainer = wrapper.find('.entry-container');

    entryContainer.trigger('click');

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: journalState.entries[0].id,
      },
    });
  });

  test('should be test computed properties', () => {
    expect(wrapper.vm.day).toBe(23);
    expect(wrapper.vm.month).toBe('July');
    expect(wrapper.vm.yearDay).toBe('2021, Friday');
  });
});
