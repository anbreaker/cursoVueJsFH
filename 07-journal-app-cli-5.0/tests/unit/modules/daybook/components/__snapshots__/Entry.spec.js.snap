// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Entry Component should be match with the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">23</span><span class="mx-2 fs-5">July</span><span class="mx-1 fw-light">2021, Friday</span></div>
  <div class="entry-description">Hello Mock Data test</div>
</div>
`;
